kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/filter/RequestFilterResultBuilderImpl.java

```diff
@@ -48,6 +50,19 @@ public CloseOrTerminalStage<RequestFilterResult> shortCircuitResponse(@NonNull A
         return this;
     }
 
+    @Override
+    public CloseOrTerminalStage<RequestFilterResult> errorResponse(RequestHeaderData header, ApiMessage request, @NonNull Throwable throwable)
+            throws IllegalArgumentException {
+        final AbstractResponse errorResponseMessage = KafkaProxyExceptionMapper.errorResponseForMessage(header, request, throwable);
+        validateShortCircuitResponse(errorResponseMessage.data());
+        final ResponseHeaderData responseHeaders = new ResponseHeaderData();
+        responseHeaders.setCorrelationId(header.correlationId());
+        this.shortCircuitHeader = responseHeaders;
+        this.shortCircuitResponse = errorResponseMessage.data();
+        this.drop();
```

Comment:
why does this path need a drop call when we don't call it for other short circuit responses like:

```
public CloseOrTerminalStage<RequestFilterResult> shortCircuitResponse(@Nullable ResponseHeaderData header, @NonNull ApiMessage message) {
        validateShortCircuitResponse(message);
        this.shortCircuitHeader = header;
        this.shortCircuitResponse = message;
        return this;
    }
```

Reply:
in the FilterHandler it looks like setting `drop` will cause it to not go down the `forwardShortCircuitResponse` path.

Reply:
Good question. 

It shouldn't

Reply:
709862be4242187e18b21f11d282831cb3df5c12